src/test/unit/app/controller/courts/AddressController.ts

```diff
@@ -0,0 +1,466 @@
+      }
+    ];
+  };
+
+  const getValidDisplayAddresses: () => DisplayCourtAddresses = () => {
+    const courtAddresses = getValidCourtAddresses();
+    const primary = courtAddresses[0];
+    const secondary = courtAddresses[1];
+    return {
+      primary: {
+        'type_id': primary.type_id,
+        'address_lines': primary.address_lines.join('\n'),
+        'address_lines_cy': primary.address_lines_cy.join('\n'),
+        town: primary.town,
+        'town_cy': primary.town_cy,
+        postcode: primary.postcode
+      },
+      secondary: {
+        'type_id': secondary.type_id,
+        'address_lines': secondary.address_lines.join('\n'),
+        'address_lines_cy': secondary.address_lines_cy.join('\n'),
+        town: secondary.town,
+        'town_cy': secondary.town_cy,
+        postcode: secondary.postcode
+      }
+    };
+  };
+
+  const addressTypes: AddressType[] = [
+    { id: 100, name: 'Visit us', 'name_cy': 'Visit us' },
+    { id: 200, name: 'Write to us', 'name_cy': 'Write to us' },
+    { id: 300, name: 'Visit or contact us' , 'name_cy': 'Visit or contact us'}
+  ];
+
+  const expectedSelectItems: SelectItem[] = [
+    {
+      value: addressTypes[0].id,
+      text: addressTypes[0].name,
+      selected: false
+    },
+    {
+      value: addressTypes[1].id,
+      text: addressTypes[1].name,
+      selected: false
+    },
+    {
+      value: addressTypes[2].id,
+      text: addressTypes[2].name,
+      selected: false
+    },
+  ];
+
+  const getExpectedResults = (
+    primaryAddress: DisplayAddress,
+    secondaryAddress: DisplayAddress,
+    expectedErrors: { text: string }[],
+    primaryPostcodeInvalid: boolean,
+    secondaryPostcodeInvalid: boolean,
+    isFatalError: boolean) => {
+    return {
+      addresses: {primary: primaryAddress, secondary: secondaryAddress},
+      addressTypesPrimary: expectedSelectItems,
+      addressTypesSecondary: [expectedSelectItems[0], expectedSelectItems[1]],
+      writeToUsTypeId: addressTypes[1].id,
+      updated: false,
+      errors: expectedErrors,
+      primaryPostcodeInvalid: primaryPostcodeInvalid,
+      secondaryPostcodeInvalid: secondaryPostcodeInvalid,
+      fatalError: isFatalError
+    };
+  };
+
+  const controller = new AddressController();
+
+  beforeEach(() => {
+    mockApi = {
+      getCourtAddresses: async (): Promise<CourtAddress[]> => getValidCourtAddresses(),
+      updateCourtAddresses: async (): Promise<CourtAddress[]> => getValidCourtAddresses(),
+      getAddressTypes: async (): Promise<AddressType[]> => addressTypes
+    };
+
+    CSRF.create = jest.fn().mockReturnValue('validCSRFToken');
+    CSRF.verify = jest.fn().mockReturnValue(true);
+
+    req.params = {
+      slug: 'central-london-county-court'
+    };
+    req.scope.cradle.api = mockApi;
+    req.scope.cradle.api.updateCourtAddresses = jest.fn().mockResolvedValue(getValidCourtAddresses());
+  });
+
+  test('Should get addresses view and render the page', async () => {
```

Comment:
Do we need one here for where there is not a secondary address? 

getExpectedResults(expectedAddresses.primary, null, [], false, false, false);